### PR TITLE
Issue #302 add ArrayTrailingComma and CommentsIndentation to CHECK_TO_ANCESTOR_NODES_MAP

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #302 add ArrayTrailingComma and CommentsIndentation to CHECK_TO_ANCESTOR_NODES_MAP

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java [hunk 1]
```diff
@@ -51,8 +51,12 @@
     private static final Map<String, List<Integer>> CHECK_TO_ANCESTOR_NODES_MAP = new HashMap<>();
 
     static {
+        CHECK_TO_ANCESTOR_NODES_MAP.put("ArrayTrailingComma",
+                Arrays.asList(TokenTypes.ARRAY_INIT));
         CHECK_TO_ANCESTOR_NODES_MAP.put("AvoidNestedBlocks",
                 Arrays.asList(TokenTypes.SLIST));
+        CHECK_TO_ANCESTOR_NODES_MAP.put("CommentsIndentation",
+                Arrays.asList(TokenTypes.METHOD_DEF));
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:59]: Comments can be anywhere - on a variable, if-statement, try-statement, class, enum, etc. 
METHOD_DEF doesn't cover all these cases, and sometimes it is bigger scope than needed. 
For example, this will trigger violations for both comments, but should be only "comment 1":
```diff
void m() {
  if (asd) {
+// comment 1
  }

  if (asd) {
// comment 2
  }
}
```

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java [hunk 2]
```diff
@@ -51,8 +51,12 @@
     private static final Map<String, List<Integer>> CHECK_TO_ANCESTOR_NODES_MAP = new HashMap<>();
 
     static {
+        CHECK_TO_ANCESTOR_NODES_MAP.put("ArrayTrailingComma",
+                Arrays.asList(TokenTypes.ARRAY_INIT));
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:55]: ArrayTrailingComma looks good. Please move it to a separate pull request, so we can merge it to master.

**Review by baratali — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:59]: @baratali is ok to use SLIST, then `if`, `try`, `method` all can be covered. then add `class` and `enum`

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 2 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:55]: @baratali ok, separate pr is here: https://github.com/checkstyle/patch-filters/pull/324

Comment (baratali) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:55]: merged

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 1 above)
Comment (baratali) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:64]: ok, let's try it. Also please add `INTERFACE_DEF` and `RECORD_DEF`

Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java]: ok, done

Comment (baratali) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java]: >  Add more tests for edge cases.

This is not done. You have so many different nodes, and only one test for a variable. Also please add example from my previous comment https://github.com/checkstyle/patch-filters/pull/321#discussion_r478568767

**Comment by baratali** [comment]
Fix commit message

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java]: @baratali add tests

**Comment by HuGanghui** [comment]
> Fix commit message Issue #302: add CommentsIndentation .... as you update only for one check

fixed.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        int a;
+            // violation without filter
+        }
+    }
+}
+
+class Test2 {
+int a;
+    // violation without filter
+}
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:22]: please add a newline at the end of file

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package TreeWalker.CommentsIndentation;
+
+public class Test {
+    public void test() {
+        try {
+        int a;
+            // violation without filter
+        } catch (Exception ex) {}
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:8]: Add another `try` into this method with a suppresed violation to show that filter covers only current SLIST

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package TreeWalker.CommentsIndentation;
+
+public class Test {
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:3]: add another method with a suppressed violation to this class to show that filter doesn't increase its scope to a CLASS_DEF.

Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:3]: Please add a comment above the class and add it to the diff:
```diff
+        // violation without a filter
public class Test {
```
This should cover CLASS_DEF scope.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    public void test() {
+        try {
+        int a;
+            // violation without filter
+        } catch (Exception ex) {}
+    }
+
+    public void test2() {
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:11]: Please add a comment above this method and add it to the diff:
```diff
+               // violation without filter
    public void test2() {
```
This should cover METHOD_DEF scope.

**Review by baratali — CHANGES_REQUESTED** [review]
Tests should have some suppressed violations as well, especially because we have a complicated Check and several ancestor nodes for it.
Also if you delete METHOD_DEF and SLIST from the nodes list, your test will not fail, but it should fail.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case4/Test.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package TreeWalker.CommentsIndentation;
+
+public interface Test {
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case4/Test.java:3]: Same as https://github.com/checkstyle/patch-filters/pull/321#discussion_r481483475
Please add a comment above this interface.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case4/Test.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+int a = 1;
+    // violation without filter
+}
+
+enum Test2 {
```
Comment (baratali) [inline src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case4/Test.java:8]: Same as https://github.com/checkstyle/patch-filters/pull/321#discussion_r481483475
Please add a comment above this enum.

**Review by baratali — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 1 above)
Comment (baratali) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java]: > Also please add example from my previous comment #321 (comment)

@HuGanghui , why didn't you add my example to the test?

Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java]: @baratali sorry, add it.

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java (hunk 3 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:22]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java (hunk 4 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:8]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java (hunk 5 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:3]: done

Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:3]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java (hunk 6 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case3/Test.java:11]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case4/Test.java (hunk 8 above)
Comment (HuGanghui) [reply src/test/resources/com/puppycrawl/tools/checkstyle/filters/suppressionjavapatchfilter/CommentsIndentation/context/case4/Test.java:10]: done

**Review by baratali — APPROVED** [review]
